src/storage/v2/inmemory/unique_constraints.cpp

```diff
@@ -302,6 +308,9 @@ InMemoryUniqueConstraints::CreateConstraint(LabelId label, const std::set<Proper
     // be removed.
     constraints_.erase(constraint);
     return ConstraintViolation{ConstraintViolation::Type::UNIQUE, label, properties};
+  } else {
+    // Add the new constraint to the optimized structure only if there are no violations.
```

Comment:
NB: This is simpler than if the new constraint were added to the structures before checking for violations. In that case, the function would have to remove the constraint from both `constraints_` and `constraints_by_label_` if a violation was found.